Typing information is incomplete/missing for storiesOf

**Describe the bug**

The index.d.ts declares this type:

```ts
export declare const storiesOf: (...args: any[]) => any;
```

 This 'breaks' code completion (in that there isnt any) and requires explicitly defining the types of the chain functions.

```
src/Exercises.stories.tsx:26:19 - error TS7006: Parameter 'getStory' implicitly has an 'any' type.

26     .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
                     ~~~~~~~~
```

`@storybook/react` is typed properly.

--- a/app/react-native/src/index.ts
+++ b/app/react-native/src/index.ts
@@ -1,17 +1,20 @@
 /* eslint-disable prefer-destructuring */
+import { ClientApi } from '@storybook/client-api';
 import Preview from './preview';
 
 const preview = new Preview();
 
-const rawStoriesOf = preview.api().storiesOf.bind(preview);
-export const setAddon = preview.api().setAddon.bind(preview);
-export const addDecorator = preview.api().addDecorator.bind(preview);
-export const addParameters = preview.api().addParameters.bind(preview);
-export const clearDecorators = preview.api().clearDecorators.bind(preview);
+const rawStoriesOf: ClientApi['storiesOf'] = preview.api().storiesOf.bind(preview);
+export const setAddon: ClientApi['setAddon'] = preview.api().setAddon.bind(preview);
+export const addDecorator: ClientApi['addDecorator'] = preview.api().addDecorator.bind(preview);
+export const addParameters: ClientApi['addParameters'] = preview.api().addParameters.bind(preview);
+export const clearDecorators: ClientApi['clearDecorators'] = preview
+  .api()
+  .clearDecorators.bind(preview);
 export const configure = preview.configure;
-export const getStorybook = preview.api().getStorybook.bind(preview);
+export const getStorybook: ClientApi['getStorybook'] = preview.api().getStorybook.bind(preview);
 export const getStorybookUI = preview.getStorybookUI;
-export const raw = preview.api().raw.bind(preview);
+export const raw: ClientApi['raw'] = preview.api().raw.bind(preview);
 
-export const storiesOf = (...args: any[]) =>
-  rawStoriesOf(...args).addParameters({ framework: 'react-native' });
+export const storiesOf = (kind: string, module: NodeModule) =>
+  rawStoriesOf(kind, module).addParameters({ framework: 'react-native' });

